.github/workflows/publish_charm.yaml

```diff
@@ -47,26 +119,39 @@ jobs:
           cp -rp $TEMP_DIR/. .
           rm -rf $TEMP_DIR
       - name: Install charmcraft
-        run: sudo snap install charmcraft --classic
-      - name: Log in to the Container registry
-        uses: docker/login-action@v2
+        run: |
+          echo ${{ needs.get-issue-number.outputs.pr-number }}
+          sudo snap install charmcraft --classic
+      - name: Download docker artifacts
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          github_token: ${{secrets.GITHUB_TOKEN}}
+          if_no_artifact_found: warn
+          # name: ${matrix.image}
+          pr: ${{ needs.get-issue-number.outputs.pr-number }}
+          workflow: build_images.yaml
+          workflow_conclusion: success
```

Comment:
I think the workflow runs are actually associated with the workflow that trigger the workflow, instead of the subordinate workflows invoked by the main workflow. For example, the integration test triggered by pull request is actually searchable by the `.github/workflows/integration_test.yaml` workflow file instead of `build_images.yaml`.

Also, we don't think we need this external action, we can directly find the workflow run using the commit sha and GitHub cli:
```
gh api \
  --method GET \
  -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  /repos/canonical/${{ github.repository }}/actions/runs \
  -f status=success \
  -f head_sha=$COMMIT_SHA \
  -f event=pull_request \
  --jq '[.workflow_runs[] | select(.path == ".github/workflows/integration_test.yaml") | .id]' | read -r RUN_IDS

[[ $(echo $RUN_IDS | jq length) -eq 0 ]] && echo no workflow found && exit 1

[[ $(echo $RUN_IDS | jq length) -gt 1 ]] && echo multiple workflows found && exit 1

RUN_ID=$(echo $RUN_IDS | jq ".[0]")

gh run download $RUN_ID -R ${{ github.repository }}
```

Reply:
I'll try that. Thanks for the input